### DISCUSSION TITLE
Additional endpoint text field

### DISCUSSION THREAD

**Discussion by Rodneyvs** [fennelLabs/fennel-service-api#60] [Q&A]
Hi there!

Could you add another text field to the endpoints? We want our users to add additional information about the white flags. Just a simple textarea would suffice.

With kind regards, 

Rodney

**Comment by MOsterhaus1** [comment]
In addition to this, is there also a possibility to give the a name to the infrastructure (a school in this case) in a signal? If not, is it possible to add to the encode/ decode endpoints? Or is this conflicting with the Whiteflag protocol?

**Comment by Romulus10** [comment]
Unfortunately the Whiteflag protocol doesn't support arbitrary text added to encoded messages (except for the dedicated Free Text signal). We can, however, add a name/text field at the API level. That would mean that information added about the signal would be visible to other people using this API, but it wouldn't be visible on chain. The other option is to generate a Free Text signal according to that message code in the Whiteflag protocol, reference the signal you want to add text to, and chain the signals together. I'm thinking having an internal text field might be a bit more straightforward, if less publicly visible. Let me know what you think and we can make that change.

**Reply by MOsterhaus1** [reply]
We discussed this, also with the people of the Whiteflag protocol. We agreed to use the Free Text signal from the protocol for both the name and the free text box. We will send both the name and free text as a stringified json object in the Free Text signal (as `"text": {"name": "Name example", "free_text": "lorem ipsum"}"`.

Is it possible that you create a new endpoint where we can send both signals (the infrastructure signal and the free text signal), and that they are coupled in the endpoint? This can reduce the number of API calls.

**Reply by Romulus10** [reply]
I absolutely can. I'll use the same model I've been using for on-chain actions - we'll have one endpoint for getting back the fees you'd pay on each transaction, then one actually carrying out the two transactions. I'll get that out in the next API version this weekend.
